src/Aspire.Hosting/Utils/ContainerReferenceParser.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Text.RegularExpressions;
+
+namespace Aspire.Hosting.Utils;
+
+/// <summary>
+/// Class to parse container references (e.g. "mcr.microsoft.com/dotnet/sdk:8.0")
+/// </summary>
+internal partial class ContainerReferenceParser
+{
+    public static ContainerReference Parse(string input)
+    {
+        if (string.IsNullOrEmpty(input))
+        {
+            throw new ArgumentOutOfRangeException(nameof(input), "repository name must have at least one component");
+        }
+
+        var match = ImageNameRegex().Match(input);
+
+        if (!match.Success)
+        {
+            throw new ArgumentOutOfRangeException(nameof(input), input, "invalid reference format");
+        }
+
+        return new(
+            GetGroupValueOrDefault(match.Groups["registry"]),
+            match.Groups["image"].Value,
+            GetGroupValueOrDefault(match.Groups["tag"]),
+            GetGroupValueOrDefault(match.Groups["digest"])
+            );
+
+        static string? GetGroupValueOrDefault(Group group)
+            => group.Success ? group.Value : default;
+    }
+
+    [GeneratedRegex("^((?<registry>((localhost|\\[(?:[a-fA-F0-9:]+)\\]|([\\w-]+(\\.[\\w-]+)+))(:\\d+)?)|([\\w-]+:\\d+))\\/)?(?<image>[\\w-./<>]+)(:(?<tag>[\\w-.<>]+))?(@(?<digest>.+))?$")]
```

Comment:
Based on https://github.com/dotnet/aspire/blob/ea985ec4fd70b9eb8cb2f8c1a567372f5475c89c/src/Aspire.Hosting/Utils/ContainerReferenceParser.cs#L40 with addition for IPv6 addresses